### PR TITLE
refactor: in generated libs move streaming update function internal

### PR DESCRIPTION
**Pull request by scotthart**
fixes: #8139

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218)

### CODE REVIEW

File: google/cloud/bigquery/bigquery_read_connection.h [hunk 1]
```diff
@@ -46,10 +46,6 @@ using BigQueryReadLimitedErrorCountRetryPolicy =
     ::google::cloud::internal::LimitedErrorCountRetryPolicy<
         bigquery_internal::BigQueryReadRetryTraits>;
 
-void BigQueryReadReadRowsStreamingUpdater(
```
Comment (coryan) [inline google/cloud/bigquery/bigquery_read_connection.h:49]: Unfortunately this library is GA.  I think we need to define a (hand-crafted) function that maintains backwards compat.  You can mark it deprecated with the macro that expands to `[[deprecated]]` (GOOGLE_CLOUD_CPP_DEPRECATED? cannot remember right now).

**Review by coryan — CHANGES_REQUESTED** [review]
We still want backwards compat for the BigQuery library.

**Review by scotthart — COMMENTED** [review]
I've added in a deprecated, backwards compatible function with a placeholder for a tracking issue. If we decide to go with this approach, I'll create the issue and update the numbers accordingly.

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-:-MuwdZCR9MrxAnJMNZGQ:b6apg49)* status: 0 of 24 files reviewed, all discussions resolved

**Review by coryan — CHANGES_REQUESTED** [review]
> Reviewed 17 of 19 files at r1, 6 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218)* status: 23 of 24 files reviewed, 1 unresolved discussion (waiting on @coryan and @scotthart)

___
*[generator/generator_config.textproto, line 96 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-MuwfErL6-8i2qk1dfbg:-MuwfErM144ISHnb5az3:b-in4tbr) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/741aedffc8c225e72f9838ea39ca06ad2c507540/generator/generator_config.textproto#L96)):*
> ```
>   initial_copyright_year: "2021"
>   omitted_services: ["BigQueryWrite"]
>   # TODO(#xxxx): Once deprecated streaming function is removed, connection
> ```

Ugh... I see.  Is there any hand-crafted header where you could insert this function instead?  Or we could change the generator to emit:

```cc
#include "google/cloud/$library$/custom.h"
```

If something like `emit_custom_include` is defined in the generator config?  I realize this is getting convoluted.  Let's talk tomorrow.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218)* status: 18 of 24 files reviewed, 1 unresolved discussion (waiting on @coryan)

___
*[generator/generator_config.textproto, line 96 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-MuwfErL6-8i2qk1dfbg:-Mv0uIB47mpD0aZPNJJT:b-h0518i) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/741aedffc8c225e72f9838ea39ca06ad2c507540/generator/generator_config.textproto#L96)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Ugh... I see.  Is there any hand-crafted header where you could insert this function instead?  Or we could change the generator to emit:

```cc
#include "google/cloud/$library$/custom.h"
```

If something like `emit_custom_include` is defined in the generator config?  I realize this is getting convoluted.  Let's talk tomorrow.
</blockquote></details>

Per our team meeting, I added BigQueryRead specific logic to generate the declaration and definition for a backward compatible streaming update function.

**Review by coryan — APPROVED** [review]
:lgtm: 

Please point the TODO entries to real bugs and cleanup any debugging statements before merging.

> Reviewed 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-:-Mv0yZxO0hnNdDWUhlS_:b-yqd9hb)* status: all files reviewed, 1 unresolved discussion (waiting on @scotthart)

___
*[ci/abi-dumps/google_cloud_cpp_bigquery.expected.abi.dump.gz, line 0 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-Mv0z-6Z5exsGbwAI7ZX:-Mv0z-6Z5exsGbwAI7ZY:b-y7gg84) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8083d6862bc3d8d8dee0a025abfb626f4192249e/ci/abi-dumps/google_cloud_cpp_bigquery.expected.abi.dump.gz#L0)):*
I expect this file can be reverted?
___
*[generator/internal/connection_generator.cc, line 87 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-Mv0yIn_2-UK2OOEEdbU:-Mv0yIna2TycmDYNzsJ-:bwl2dk1) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8083d6862bc3d8d8dee0a025abfb626f4192249e/generator/internal/connection_generator.cc#L87)):*
> ```cpp
>   // TODO(#xxxx): This is a special case for backwards compatibility of the
>   // streaming updated function.
>   std::cout << "serviceme = " << vars().at("service_name") << "\n";
> ```

This looks like a debugging statement?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218)* status: 21 of 24 files reviewed, 1 unresolved discussion (waiting on @coryan)

___
*[generator/internal/connection_generator.cc, line 87 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-Mv0yIn_2-UK2OOEEdbU:-Mv19KjSAaT4oAEZ4aDw:ba3dfob) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8083d6862bc3d8d8dee0a025abfb626f4192249e/generator/internal/connection_generator.cc#L87)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

This looks like a debugging statement?
</blockquote></details>

Removed
___
*[ci/abi-dumps/google_cloud_cpp_bigquery.expected.abi.dump.gz, line  at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8218#-Mv0z-6Z5exsGbwAI7ZX:-Mv1BSjs8FzQyHlNjksb:bvth56w) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8083d6862bc3d8d8dee0a025abfb626f4192249e/ci/abi-dumps/google_cloud_cpp_bigquery.expected.abi.dump.gz#L)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

I expect this file can be reverted?
</blockquote></details>

reverted

**Pull request merged**